tests/modules/commandfilter.c

```diff
@@ -238,12 +252,16 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
                 REDISMODULE_CMDFILTER_NOSELF)) == NULL)
         return REDISMODULE_ERR;
 
+    module_ctx = RedisModule_GetDetachedThreadSafeContext(ctx);
```

Comment:
I don't think this is a valid use of a thread safe context (main thread and doesn't lock the GIL).

let's continue the discussion in https://github.com/redis/redis/pull/12680#discussion_r1384931162
@MeirShpilraien maybe you share your view.

Reply:
Its a detached ctx, suppose to be mostly use for logs. I see you are using it inside the post filter callback you added. I think that if our intention is to allow to touch the keyspace on this callback then we should provide it with a context. Providing a `RedisModuleCommandFilterCtx` indicate that it should not touch the keyspace as the filter docs says:
```
* The filter executes in a special filter context, which is different and more
 * limited than a RedisModuleCtx.  Because the filter affects any command, it
 * must be implemented in a very efficient way to reduce the performance impact
 * on Redis.  All Redis Module API calls that require a valid context (such as
 * `RedisModule_Call()`, `RedisModule_OpenKey()`, etc.) are not supported in a
 * filter context.
```

Also I wonder whats the use-case for which you need this new API? Maybe key space notification + post notification job can fulfill it? I am asking because I can see different cases that can go wrong here, for example, what if a module invoke a read command from a key space notification which will call the filter that will change the data and will violate the read only constrains of key space notifications? I think that if we conclude that this new API is really needed, then it should be a read only callback (just like key space notifications) and the write operations should be done on a post execution job.